File: 0001-remove-duplicates-in-sorted-SLL/RemoveDuplicates.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode currentNode = head;
        while(currentNode!=null){
            ListNode nextDistinctNode = currentNode.next;
            while(nextDistinctNode!=null && nextDistinctNode.val == currentNode.val){
                nextDistinctNode = nextDistinctNode.next;
            }
            currentNode.next = nextDistinctNode;
            currentNode = nextDistinctNode;
        }
        return head;
    }
}

File: 0001-remove-duplicates-in-sorted-SLL/RemoveDuplicates.cs

public class Solution
{
    public class LinkedList
    {
        public int value;
        public LinkedList next;
        public LinkedList(int value){
            this.value = value;
            this.next = null;
        }
    }
    public LinkedList RemoveDuplicatesFromLinkedList(Linkedlist head){
        LinkedList currentNode = head;
        while(currentNode!=null){
            LinkedList nextDistinctNode = currentNode.next;
            while(nextDistinctNode!=null && nextDistinctNode.value == currentNode.value){
                nextDistinctNode = nextDistinctNode.next;
            }
            currentNode.next = nextDistinctNode.next;
            currentNode = nextDistinctNode;
        }
        return head;
    }
}